rfc/router-storage.md

```diff
@@ -0,0 +1,767 @@
+
+2. Invoke `blob/accept` on storage node
+
+```json
+ // 3. System will attempt to accept uploaded content that matches blob
+  // multihash and size.
+  { // "/": "bafy...accept",
+
+    "iss": "did:web:web3.storage",
+    "aud": "did:web:node1.storage",
+    "cmd": "/service/blob/accept",
+    "sub": "did:web:web3.storage",
+    "args": {
+      // as best I can tell this is used to determine who is the audience for the location commitment, which should be web3.storage
+      "space": "did:web:web3.storage",
+      "blob": {
+        // multihash of the blob as byte array
+        "content": { "/": { "bytes": "mEi...sfKg" } },
+        "size": 2_097_152,
+      },
+      "expires": 1712903125,
+    }
+  }
+```
+
+Legacy Invocation:
+
+```json
+{ // "/": "bafy...accept",
+  "iss": "did:web:web3.storage",
+  "aud": "did:web:node1.storage",
+  "att": {
+    "can": "/web3.storage/blob/accept",
+    "with": "did:web:web3.storage",
+    "nb": {
+      // as best I can tell this is used to determine who is the audience for the location commitment, which should be web3.storage?
+      // alternatively maybe it should be directly to space
+      "space": "did:web:web3.storage",
+      "blob": {
+        // multihash of the blob as byte array
+        "content": { "/": { "bytes": "mEi...sfKg" } },
+        "size": 2_097_152,
+      },
+      "expires": 1712903125,
+    }
+  }
+}
+```
+
+Receipt:
+
+```json
+{ 
+  // "/": "bafy..work",
+  "iss": "did:web:node1.storage",
+  "aud": "did:web:web3.storage",
+  "cmd": "/ucan/assert",
+  "sub": "did:web:web3.storage",
+  "args": {
+    "assert": [
+      // refers to the invocation from the example
+      { "/": "bafy..accept" },
+      // refers to the receipt corresponding to the above invocation
+      { 
+        "out": {
+          "ok": {
+            // result is commitment that is produced as result of "bafy..accept"
+            "site": "bafy..locationCommitment"
+          }
+        },
+        "next": []
+      },
+    ],
+  },
+}
+```
+
+Legacy receipt:
+
+```json
+{ 
+  // "/": "bafy..work",
+  "iss": "did:web:node1.storage",
+  "aud": "did:web:web3.storage",
+  "ran": {
+    "/": "bafy..accept"
+  },
+  "out": {
+    "ok": {
+      // result is commitment that is produced as result of "bafy..accept"
+      "site": "bafy..locationCommitment"
+    }
+  },
+  "fx": [],
+}
+```
+
+4. Router fetches location commitment, and creates new commitment for the user.
+
+**Note: initially I thought this was a redelegation, but I think it's maybe not -- the commitment here is to a limited download URL of JUST the blob -- it's not a w3s gateway url with full download capabilities.
+
+the "did:web:web3.storage" is maybe really a did for two seperate things:
+- the web3.storage read pipeline
+- the web3.storage write pipeline
+
+As long as we're the only ones running the read pipeline, it can all go through w3s.link, but I imagine a future situation where other people want to use another read pipeline provider.
```

Comment:
Exactly, which is why I think user can delegate location commitment to say `did:web:w3s.link` authorizing it to serve this content at their expense, but they also could go and authorize other gateways or libp2p nodes for different read pipelines.

Reply:
makes sense